src/cc/frontends/clang/kbuild_helper.cc

```diff
@@ -107,4 +107,54 @@ int KBuildHelper::get_flags(const char *uname_machine, vector<string> *cflags) {
   return 0;
 }
 
+int proc_kheaders_exists(void)
+{
+  struct stat buffer;
+  return (stat(PROC_KHEADERS_PATH, &buffer) == 0);
+}
+
+int extract_kheaders(std::shared_ptr<ebpf::TmpDir> &tmpdir)
+{
+  char tar_cmd[256];
+  int ret;
+  bool module = false;
+
+  if (!proc_kheaders_exists()) {
+    ret = system("modprobe kheaders");
+    if (ret)
+      return ret;
+    if (!proc_kheaders_exists())
+      return -1;
+    module = true;
+  }
+
+  ret = system("rm -rf /tmp/proc-headers-*");
```

Comment:
what happens when multiple bcc scripts run at the same time? a race to extract/rm ?
shouldn't all bcc scripts reuse single dir?

Reply:
yes you're right, I missed that. They could race. I will try such scenarios and fix, thanks

Reply:
I will come with an approach that tries to reuse a single directory, but in the worst case could have  different directories if there are multiple processes trying to do extraction at the same time. In the most common case, only a single directory will be used even for racing tasks. Since its only a single /tmp directory now, clean up will be left to be handled automatically by the system (such as reboot).  Does that sound Ok to you?